libdrgn/program.c

```diff
@@ -673,14 +669,19 @@ drgn_program_load_debug_info(struct drgn_program *prog, const char **paths,
 		drgn_dwarf_index_report_abort(dindex);
 		return err;
 	}
-	report_from_dwfl = (!(prog->flags & DRGN_PROGRAM_IS_LINUX_KERNEL) &&
-			    load_main);
+	report_from_dwfl = (!is_kernel && load_main);
 	err = drgn_dwarf_index_report_end(dindex, report_from_dwfl);
 	if ((!err || err->code == DRGN_ERROR_MISSING_DEBUG_INFO) &&
 	    !prog->has_platform) {
 		dwfl_getdwarf(prog->_dicache->dindex.dwfl,
 			      drgn_set_platform_from_dwarf, prog, 0);
 	}
+
+	if (!is_kernel && !prog->lang) {
+		err = drgn_program_set_language_from_main(prog);
```

Comment:
I think there was a bug where this was running too soon (even if main was available, it would not be found). Moving the invocation here fixes it, but I'm not sure if I kept the semantics of the original call.

Reply:
Yup, this seems to work, thanks so much for fixing that! I'll rebase minus my first commit on that branch.